src/ZoneServer/World/Actors/CombatEntities/Components/MovementComponent.cs

```diff
@@ -404,13 +404,23 @@ private void CheckWarp()
 		/// <param name="type"></param>
 		public void SetMoveSpeedType(MoveSpeedType type)
 		{
-			if (this.Entity is Mob mob && this.MoveSpeedType != type)
+			if (this.MoveSpeedType == type)
+				return;
+
+			if (this.Entity is Mob mob)
 			{
 				this.MoveSpeedType = type;
 				this.Entity.Properties.Invalidate(PropertyName.MSPD);
 
 				Send.ZC_MSPD(this.Entity);
 			}
+			else if (this.Entity is Character character && character is DummyCharacter dummyCharacter)
```

Comment:
I would presume this check to be either incorrect or redundant. Am I right in that assessment? You're checking whether the entity is a character _and_ a dummy character, which would mean the code applies only to dummy characters, as only they are both. Though in that case it would be enough to check for the dummy character type. And if you wanted it to apply to both, a check for character would be enough, as the dummy inherits character iirc.